contracts/ExitFormat.sol

```diff
@@ -137,8 +153,53 @@ library ExitFormat {
                 (bool success, ) = destination.call{value: amount}(""); //solhint-disable-line avoid-low-level-calls
                 require(success, "Could not transfer ETH");
             } else {
-                // TODO support other token types via the singleAssetExit.metadata field
-                ERC20Interface(asset).transfer(destination, amount);
+                if (
+                    // ERC20 Token
+                    singleAssetExit.tokenMetadata.assetType == AssetType.ERC20
+                ) {
+                    IERC20(asset).transfer(destination, amount);
+                } else if (
+                    // ERC721 Token
+                    singleAssetExit.tokenMetadata.assetType == AssetType.ERC721
+                ) {
+                    require(amount == 1, "Amount must be 1 for an ERC721 exit");
+                    uint256 tokenId =
+                        abi
+                            .decode(
+                            singleAssetExit
+                                .tokenMetadata
+                                .metadata,
+                            (TokenIdExitMetadata)
+                        )
+                            .tokenId;
+                    IERC721(asset).safeTransferFrom(
+                        address(this),
+                        destination,
+                        tokenId
+                    );
+                } else if (
+                    // ERC1155 Token
+                    singleAssetExit.tokenMetadata.assetType == AssetType.ERC1155
+                ) {
+                    uint256 tokenId =
+                        abi
+                            .decode(
+                            singleAssetExit
+                                .tokenMetadata
+                                .metadata,
+                            (TokenIdExitMetadata)
+                        )
+                            .tokenId;
+                    IERC1155(asset).safeTransferFrom(
+                        address(this),
+                        destination,
+                        tokenId,
+                        amount,
+                        singleAssetExit.allocations[j].metadata // the metadata from the allocation is passed to the safeTransferFrom call
+                    );
+                } else {
+                    revert("unsupported token");
```

Comment:
Let's try and keep a consistent terminology, I suggest using "asset" wherever appropriate (instead of "token").

Reply:
Done